### DISCUSSION TITLE
建议功能性按钮不要作为 Schema properties 的方式去配置

### DISCUSSION THREAD

**Discussion by smalldragonluo** [alibaba/formily#1540] [Ideas]
![image](https://user-images.githubusercontent.com/7576698/120479180-3b4e2600-c3e0-11eb-9f84-4f223968ad85.png)

如图，建议功能性按钮不要作为 Schema properties 的方式去配置～ 一是有一定理解/使用成本，增大了 Schema 的体积，二来将一些不必要的底层实现暴露给用户似乎也不太优雅。是否可以用 x-component-props 去开启/配置功能呢？

**Comment by janryWang** [comment]
很早就考虑过这个了，作为properties的考虑主要是需要支持按钮的联动，比如某些字段会控制操作按钮的显示隐藏等，所以不会考虑内置功能按钮，如果用户希望内置，可以自定义组件

**Reply by smalldragonluo** [reply]
明白了，其实看起来 Formily 协议层面已经是一个页面级的渲染协议，不局限于表单，似乎后续也可能发展面向中后台的低代码搭建。不过我使用起来有一个感觉是这两者有点杂糅在一起，properties 在用户心智上是一个表单 field 字段，但渲染层借此实现了某些功能联动，是否可以考虑扩展一下支持 x-component-props 的联动，或者独立协议 🤗 （只是提一个想法，可以不用回复）
